hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -385,6 +386,10 @@ protected void completeClustering(HoodieReplaceCommitMetadata metadata, JavaRDD<
     } else {
       writeTimer = metrics.getDeltaCommitCtx();
     }
+    latestWriteInstantCompletedBeforeWriter.set(table.getMetaClient().getActiveTimeline().getAllCommitsTimeline()
```

Comment:
so, I was expecting we will do some kind of atomic swap, not just a set. if you just want to set, even a volatile variable is fine right?

Reply:
That was the intention, missed this, fixed now